src/ray/gcs/tables.h

```diff
@@ -606,6 +606,11 @@ class ClientTable : private Log<UniqueID, ClientTableData> {
   /// \return Whether the client with ID client_id is removed.
   bool IsRemoved(const ClientID &client_id) const;
 
+  /// Get the information of all clients.
+  ///
+  /// \return The clientId to Client information map.
+  const std::unordered_map<ClientID, ClientTableDataT> &GetAllClients() const;
```

Comment:
In general we should avoid exposing internal data structures, though in this case it seems like we do need a way to iterate over the clients..

Reply:
Yes... Is it possible to make it a private or protected function and make object_manager a friend to access this function?